huaweicloud/services/apig/resource_huaweicloud_apig_application_authorization.go

```diff
@@ -0,0 +1,328 @@
+package apig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/appauths"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceAppAuth() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAppAuthCreate,
+		ReadContext:   resourceAppAuthRead,
+		UpdateContext: resourceAppAuthUpdate,
+		DeleteContext: resourceAppAuthDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAppAuthImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the application and APPCODEs are located.",
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the dedicated instance to which the application and APIs belong.",
+			},
+			"application_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the application authorized to access the APIs.",
+			},
+			"env_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The environment ID where the APIs were published.",
+			},
+			"api_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The authorized API IDs",
+			},
+		},
+	}
+}
+
+func resourceAppAuthCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+	var (
+		appId  = d.Get("application_id").(string)
+		envId  = d.Get("env_id").(string)
+		apiIds = utils.ExpandToStringListBySet(d.Get("api_ids").(*schema.Set))
```

Comment:
The variable `cfg` and `region` can also be defined in `var()`

Reply:
if the client generated failed, these variables shouldn't be created.